runtime/lua/vim/lsp.lua

```diff
@@ -232,6 +232,12 @@ local function validate_client_config(config)
     flags           = { config.flags, "t", true };
     get_language_id = { config.get_language_id, "f", true };
   }
+  assert(
+    (not config.flags
```

Comment:
Why are we checking if config.flags is not nil?

Reply:
`validate_client_config` is called before we do `config.flags = config.flags or {}`, so at this point `conflig.flags` can still be `nil` which is allowed here - the assert should only trigger if flags are present and there is a non numeric `debounce_did_change_notify` property